kernel/CMakeLists.txt

```
@@ -30,16 +30,10 @@ zephyr_syscall_header_ifdef(
   ${ZEPHYR_BASE}/include/zephyr/kernel/mm/demand_paging.h
 )
 
-# If a pre-built static library containing kernel code exists in
-# this directory, libkernel.a, link it with the application code
-# instead of building from source.
-zephyr_library_get_current_dir_lib_name(${ZEPHYR_BASE} libkernel_stem)
-set(libkernel ${CMAKE_CURRENT_SOURCE_DIR}/lib${libkernel_stem}${CMAKE_STATIC_LIBRARY_SUFFIX})
-unset(libkernel_stem)
+get_target_property(KERNEL_INTERFACE kernel KERNEL_INTERFACE)
```

now that we import the kernel lib the proper way, then let's use the CMake built-in IMPORTED property instead.
```suggestion
get_target_property(is_imported kernel IMPORTED)
```

---

Nice cleanup. I've pushed this update.